src/main/java/net/openhft/chronicle/core/threads/ThreadLock.java

```diff
@@ -0,0 +1,30 @@
+package net.openhft.chronicle.core.threads;
+
+public interface ThreadLock {
+
+    /**
+     * Try to lock once or fail
+     *
+     * @param threadId to attempt to lock for
+     * @return true if the lock could be obtains, or false if not.
+     * @throws IllegalStateException if the same threadId already holds the lock
+     */
+    boolean tryLock(int threadId) throws IllegalStateException;
```

Comment:
It is very unusual and generally discouraged to declare runtime exceptions in libraries. I think this is something we should move away from. There is already a JavaDoc tag for this here.